department-of-veterans-affairs/caseflow
# Check for non nill attyid when pulling all judges

Resolves issue were non judge appears in a dropdown that should only contain judges (and acting judges). See investigation here: https://hackmd.io/@hschallh/By1eoWCO8
>Long story shorter:
> - When we pull a bulk list of judges from vacols (how we populate the dropdown), we only look at the user’s svlj  value of their vacols staff record.
> - When we check if someone is an acting judge in caseflow, we look at the svlj  AND sattyid of their vacols staff record.

From the board: All attorneys, judges, and acting judges should have a non nil sattyid

### Description
Create some convenience scopes and only return users with non nil sattyid when pulling all judges.

### Acceptance Criteria

### Testing Plan
1. Remove the sattyid from a judge's staff record
```ruby
user = FactoryBot.create(:user, css_id: "NOT_A_JUDGE", station_id: 101)
FactoryBot.create(:staff, svlj: "A", sattyid: nil, user: user)
```
2. Ensure they are not returned in Judge.list_all
```ruby
judges_for_dropdown = Judge.list_all
judges_for_dropdown.include? user
=> false
# Check for aaaaabshire just in case
judges_for_dropdown.include? User.find(3)
=> true
```
### User Facing Changes
- [ ] Non judges will no longer appear in judge only dropdowns
